docs(accessibility-design): Adds new documentation to support content audit. (#3781)
patternfly/patternfly-org: packages/documentation-site/patternfly-docs/content/developer-resources/accessibility-design.md

---
id: Design for accessibility
section: accessibility
---

As described in our [accessibility fundamentals](/accessibility/accessibility-fundamentals), users may interact with your product through a variety of assistive technologies. In addition to [developing for accessibility](/accessibility/accessibility-development), you must also incorporate it into your product's UI design. 

## Color and contrast

Color and contrast go hand-in-hand for visual design accessibility. 

For example, you should consider contrast when using colors commonly affected by color perception deficiency (such as those described in [color perception by W3C](https://www.w3.org/WAI/GL/low-vision-a11y-tf/wiki/Overview_of_Low_Vision#Color_Perception)). Different colors in a UI should be distinguishable for users across the color perception deficiency spectrum, typically by ensuring that each color has a different contrast weight. 

Designing with contrast in mind also benefits users with light sensitivity.

> [PatternFly's colors](/design-foundations/colors) pass color contrast tests, so long as you follow our recommendations. 

Reviewer: I'm wondering if we should tweak this line to be a little clearer. In our components, we've tested contrast and are good, but I worry to just say that "PF colors pass color contrast" because you can still use PF colors in a way that does not pass color contrast. That's pretty much true with any color palette, for example [slide 40 of this a11y design presentation](https://docs.google.com/presentation/d/1xomLQRUJ2IJR4n-tRyK13B09G2IkcMGXy1p8_OapDcI/edit#slide=id.ge5933efd32_1_3392) shows the RH color palette with the different combinations of colors that wouldn't be accessible. Maybe we can say something like, "[PatternFly's colors](/design-foundations/colors) pass color contrast tests in our components, so long as you follow our recommendations. If you combine PF colors in other ways, please check the color contrast with a contrast checker"?

Designing with contrast in mind also benefits users with light sensitivity.

[PatternFly's colors](/design-foundations/colors) pass color contrast tests, so long as you follow our recommendations. 

## Content 

Your text content should also consider the needs of users with low vision. For example, links should be descriptive and concise, so that users can quickly identify their purpose. 

To help simplify and strengthen the way that users process content, it is also important to carefully consider your text size, font, letter spacing, margins, and spacing. PatternFly supports these considerations through our [typography](/design-foundations/typography) and [spacing](/design-foundations/spacers) guidelines.

## Understanding low vision 

In addition to following the previous recommendations, it can be helpful to develop an understanding of low-vision users, so that their needs are top of mind as you design.

**Low vision** is frequently defined as a visual impairment that impedes a person’s ability to perform everyday tasks, which cannot be corrected with glasses, contacts, medicine, or surgery. 

There are 4 types of limitations as a result of low vision.

> ### 1. Visual acuity 

Reviewer 2: Nit but not a blocker, similar to another comment from one of the other PRs but wondering if we need the numbers for these 4 sections

Author: yep - good call!

---

> id: Accessible design

Author: ```suggestion
id: Design accessibly
```

> id: Design accessibly

Author: ```suggestion
id: Design for accessibility
```

section: accessibility
---

> As described in our [accessibility fundamentals](/accessibility/accessibility-fundamentals), users may interact with your product through a variety of assistive technologies. In addition to [developing accessibly](/accessibility/accessibility-development), you must design accessibly. 

Author: ```suggestion
As described in our [accessibility fundamentals](/accessibility/accessibility-fundamentals), users may interact with your product through a variety of assistive technologies. In addition to [developing for accessibility](/accessibility/accessibility-development), you must also incorporate it into your product's UI design. 
```

> 

Author: ```suggestion
The following table differentiates the accessibility responsibilities that are up to designers, developers, or both to uphold.
```

> The following table differentiates the accessibility responsibilities that are up to designers, developers, or both to uphold.

Author: ```suggestion

The following table differentiates the accessibility responsibilities that are up to designers, developers, or both to uphold.
```

| Designer | Designer and developer | Developer |
| --- | --- | --- |
| Color  | Layout and structure  | Semantic HTML  |

> | Contrast  | Information architecture and navigation | tab order  |

Author: ```suggestion
| Contrast  | Information architecture and navigation | Tab order  |
```

| Content  | Interaction patterns | Bypass blocks |

## Color and contrast

> Color and contrast go hand-in-hand for visual design. 

Author: ```suggestion
Color and contrast go hand-in-hand for visual design accessibility. 
```

> | Designer | Designer and developer | Developer |
> | --- | --- | --- |
> | Color  | Layout and structure  | Semantic HTML  |
> | Contrast  | Information architecture and navigation | Tab order  |
> | Content  | Interaction patterns | Bypass blocks |

Author: ```suggestion

| Designer | Designer and developer | Developer |
| --- | --- | --- |
| Color  | Layout and structure  | Semantic HTML  |
| Contrast  | Information architecture and navigation | Tab order  |
| Content  | Interaction patterns | Bypass blocks |
```

For example, you should consider contrast when using colors commonly affected by color perception deficiency ([such as those described by W3C](https://www.w3.org/WAI/GL/low-vision-a11y-tf/wiki/Overview_of_Low_Vision#Color_Perception)). Different colors in a UI should be distinguishable for users across the color perception deficiency spectrum, typically by ensuring that each color has a different contrast weight. 

Designing with contrast in mind also benefits users with light sensitivity.

[PatternFly's colors](/design-foundations/colors) pass color contrast tests, so long as you follow our recommendations. 

## Content 

Your text content should also consider the needs of users with low vision. For example, links should be descriptive and concise, so that users can quickly identify their purpose. 

> To help simplify and strengthen the way that users process content, it is also important to carefully consider your text size, font, letter spacing, margins, and spacing. PatternFly supports these considerations through its [typography](/design-foundations/typography) and [spacing](/design-foundations/spacers) guidelines.

Author: ```suggestion
To help simplify and strengthen the way that users process content, it is also important to carefully consider your text size, font, letter spacing, margins, and spacing. PatternFly supports these considerations through our [typography](/design-foundations/typography) and [spacing](/design-foundations/spacers) guidelines.
```

---
id: Design for accessibility
section: accessibility
---

As described in our [accessibility fundamentals](/accessibility/accessibility-fundamentals), users may interact with your product through a variety of assistive technologies. In addition to [developing for accessibility](/accessibility/accessibility-development), you must also incorporate it into your product's UI design. 

> The following table differentiates the accessibility responsibilities that are up to designers, developers, or both to uphold.

Author: ```suggestion
```

> | Designer | Designer and developer | Developer |
> | --- | --- | --- |
> | Color  | Layout and structure  | Semantic HTML  |
> | Contrast  | Information architecture and navigation | Tab order  |
> | Content  | Interaction patterns | Bypass blocks |

Author: ```suggestion
```

## Color and contrast

Color and contrast go hand-in-hand for visual design accessibility. 

> For example, you should consider contrast when using colors commonly affected by color perception deficiency ([such as those described by W3C](https://www.w3.org/WAI/GL/low-vision-a11y-tf/wiki/Overview_of_Low_Vision#Color_Perception)). Different colors in a UI should be distinguishable for users across the color perception deficiency spectrum, typically by ensuring that each color has a different contrast weight. 

Reviewer 2: Nit: could we update the link text here? Just out of context, there's the question of "what are those?" in regards to the "such as those..." text. Maybe wrapping the link around the "color perception" text instead, or "(as described in [color perception by W3C](...)"?

> For example, you should consider contrast when using colors commonly affected by color perception deficiency ([such as those described by W3C](https://www.w3.org/WAI/GL/low-vision-a11y-tf/wiki/Overview_of_Low_Vision#Color_Perception)). Different colors in a UI should be distinguishable for users across the color perception deficiency spectrum, typically by ensuring that each color has a different contrast weight. 

Author: ```suggestion
For example, you should consider contrast when using colors commonly affected by color perception deficiency (such as those described in [color perception by W3C](https://www.w3.org/WAI/GL/low-vision-a11y-tf/wiki/Overview_of_Low_Vision#Color_Perception)). Different colors in a UI should be distinguishable for users across the color perception deficiency spectrum, typically by ensuring that each color has a different contrast weight. 
```
